server/shared/storage/storage.controller.js

```diff
@@ -3,22 +3,56 @@
 const { getFileUrl, saveFile } = require('./');
 const { readFile, sha256 } = require('./util');
 const config = require('../../../config/server').storage;
+const fecha = require('fecha');
+const fromPairs = require('lodash/fromPairs');
+const JSZip = require('jszip');
+const mime = require('mime-types');
 const path = require('path');
+const pickBy = require('lodash/pickBy');
 
 function getUrl(req, res) {
   const { query: { key } } = req;
   return getFileUrl(key).then(url => res.json({ url }));
 }
 
-async function upload({ file }, res) {
+async function upload({ file, body, user }, res) {
+  const { name } = path.parse(file.originalname);
+  if (body.unpack) {
+    const timestamp = fecha.format(new Date(), 'YYYY-MM-DDTHH:mm:ss');
+    const root = `${timestamp}__${user.id}__${name}`;
```

Comment:
Some components expect old schema to reconstruct the original filename (lms). Let's try to keep the old schema for nonarchived files if possible.